pkg/mev/bloxroute_submit_bundle.go

```diff
@@ -14,69 +14,76 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
-type Builder string
+type BlxrBuilder string
 
 const (
-	BuilderBloxroute    Builder = "bloxroute"
-	BuilderFlashbot     Builder = "flashbots"
-	BuilderBeaverBuild  Builder = "beaverbuild"
-	BuilderRsyncBuilder Builder = "rsync-builder"
-	BuilderAll          Builder = "all"
+	BuilderBloxroute    BlxrBuilder = "bloxroute"
+	BuilderFlashbot     BlxrBuilder = "flashbots"
+	BuilderBeaverBuild  BlxrBuilder = "beaverbuild"
+	BuilderRsyncBuilder BlxrBuilder = "rsync-builder"
+	BuilderAll          BlxrBuilder = "all"
 )
 
-// BloxrouteSubmitBundle https://docs.bloxroute.com/apis/mev-solution/bundle-submission
-func BloxrouteSubmitBundle( // nolint: cyclop
-	ctx context.Context, c *http.Client, auth, endpoint string,
-	param *BLXRSubmitBundleParams, options ...BloxrouteSubmitBundleOption,
-) (SendBundleResponse, error) {
-	var opts blxrSubmitBundleOptions
-	for _, fn := range options {
-		if fn == nil {
-			continue
-		}
-		fn(&opts)
-	}
+type BloxrouteBundleSender struct {
+	c               *http.Client
+	endpoint        string
+	auth            string
+	flashbotKey     *ecdsa.PrivateKey
+	enabledBuilders []BlxrBuilder
+}
 
-	mevBuilders := make(map[Builder]string)
-	if opts.builderBloxroute {
-		mevBuilders[BuilderBloxroute] = ""
+// NewBloxrouteBundleSender set flashbotKey to nil if you don't want to send to flashbot builders
+// With BuilderAll still need to add the flashbot key & the flashbot builder separately
+// https://docs.bloxroute.com/apis/mev-solution/bundle-submission
+func NewBloxrouteBundleSender(
+	c *http.Client,
+	endpoint, auth string,
+	flashbotKey *ecdsa.PrivateKey,
+	enabledBuilders ...BlxrBuilder,
+) *BloxrouteBundleSender {
+	return &BloxrouteBundleSender{
+		c:               c,
+		endpoint:        endpoint,
+		auth:            auth,
+		flashbotKey:     flashbotKey,
+		enabledBuilders: enabledBuilders,
 	}
-	if opts.builderFlashbot != nil {
-		sig, err := bloxrouteSignFlashbot(opts.builderFlashbot, param)
-		if err != nil {
-			return SendBundleResponse{}, fmt.Errorf("sign flashbot error: %w", err)
+}
+
+func (s *BloxrouteBundleSender) Send(
+	ctx context.Context, blockNumber uint64, txs ...*types.Transaction,
+) (SendBundleResponse, error) {
+	p := new(BLXRSubmitBundleParams).SetBlockNumber(blockNumber).SetTransactions(txs...)
+
+	mevBuilders := make(map[BlxrBuilder]string)
+	for _, b := range s.enabledBuilders {
+		if b == BuilderFlashbot && s.flashbotKey != nil {
```

Comment:
too adhoc, other builder don't need key?
I think can init a map builder with enable auth?

Reply:
Only the flashbot builder requires a key 🤔, and the signature is generated for each request. So creating a map still requires inserting the key for the flashbot builder separately